configuration/secrets.rst

```diff
@@ -0,0 +1,319 @@
+                http://symfony.com/schema/dic/doctrine
+                https://symfony.com/schema/dic/doctrine/doctrine-1.0.xsd">
+
+            <doctrine:config>
+                <doctrine:dbal
+                    password="%env(secret:DATABASE_PASSWORD)%"
+                />
+            </doctrine:config>
+
+            <parameters>
+                <parameter key="database_password">not a secret</parameter>
+            </parameters>
+
+        </container>
+
+    .. code-block:: php
+
+        // config/packages/doctrine.php
+        $container->loadFromExtension('doctrine', [
+            'dbal' => [
+                'password' => '%env(secret:DATABASE_PASSWORD)%',
+            ]
+        ]);
+        $container->setParameter('database_password', 'not a secret');
+
+Then overriding it in production environment:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/prod/doctrine.yaml
+        parameters:
+            database_password: '%env(secret:DATABASE_PASSWORD)'
+
+    .. code-block:: xml
+
+        <!-- config/packages/prod/doctrine.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:doctrine="http://symfony.com/schema/dic/doctrine"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                https://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/doctrine
+                https://symfony.com/schema/dic/doctrine/doctrine-1.0.xsd">
+
+            <parameters>
+                <parameter key="database_password">%env(secret:DATABASE_PASSWORD)</parameter>
+            </parameters>
+
+        </container>
+
+    .. code-block:: php
+
+        // config/packages/prod/doctrine.php
+        $container->setParameter('database_password', '%env(secret:DATABASE_PASSWORD)');
+
+.. _secrets-list:
+
+List existing secrets
+---------------------
+
+Everybody is allowed to list the secret's name with the command ``secrets:list``.
+If you have the ``encryption key`` you can also reveal the plain text value by
+passing the optoin ``--reveal`` to the command
+
+.. code-block:: terminal
+
+    $ APP_ENV=prod php bin/console secrets:list --reveal
+
+     ------------------- ------------------
+      key                 plaintext secret
+     ------------------- ------------------
+      DATABASE_PASSWORD   my-secret
+     ------------------- ------------------
+
+.. _secrets-deploy
+
+Deploy secret to production
+---------------------------
+
+As the ``encryption key`` is not commited, during development, you'll have to
+manualy deploy the key (once and for all) at the path referenced in the
+``encryption_key`` configuration key. Default is ``%kernel.project_dir%/config/secrets/%kernel.environment%``.
+
+.. _secrets-custom-storage
+
+Custom Secret Storage
```

Comment:
Are you not afraid that this kind of implementation can be wrong and so leak sensitive data?

Reply:
IMHO user a free to implement their own business. Even if it break things. 

Should we disallow user to run SQL queries in DBAL because they'll be able to run a `drop database`?

Reply:
I see your point on this, but maybe need to document that this kind of implementation is "risky" or something like that at least, wdyt?